clang/lib/Sema/SemaChecking.cpp

```diff
@@ -5272,8 +5272,11 @@ bool Sema::BuiltinAssumeAligned(CallExpr *TheCall) {
   {
     ExprResult FirstArgResult =
         DefaultFunctionArrayLvalueConversion(FirstArg);
-    if (checkBuiltinArgument(*this, TheCall, 0))
```

Comment:
checkBuiltinArgument should always either produce an expression with the correct type, or error out.  If neither is happening, there's something wrong with type-checking.

Reply:
Took another look at this... probably this is okay.  checkBuiltinArgument isn't actually doing anything useful here, and other places do something similar with DefaultFunctionArrayLvalueConversion().

The comment about "In-place updation of FirstArg by checkBuiltinArgument" should be deleted, though.

Reply:
Done